mod/state-transition/pkg/core/state_processor_payload.go

```diff
@@ -51,7 +51,13 @@ func (sp *StateProcessor[
 	payloadTimestamp := payload.GetTimestamp().Unwrap()
 	consensusTimestamp := ctx.GetConsensusTime().Unwrap()
 
-	sp.metrics.gaugeTimestamps(payloadTimestamp, consensusTimestamp)
+	// add telemetry for the diff between payloadTimestamp and
+	// consensusTimestamp
+	err := sp.metrics.gaugePayloadConsensusTimestampDiff(
+		payloadTimestamp, consensusTimestamp)
```

Comment:
How about instead of all of this we go though time.Duration?
Turn the timestamp into a time.Time objects and get diff as TimeA.Sub(TimeB)?